src/cli/cli.ts

```diff
@@ -32,7 +32,8 @@ async function main(): Promise<void> {
   cli.register(PushMarketing)
 
   const args = process.argv.slice(2)
-  await cli.runExit(args, context)
+  process.exitCode = await cli.run(args, context)
+  await connections.amqpClient.close()
```

Comment:
Idea: You can create `closeConnections(connections: DbConnections)` function to close all connected resources. In there you can either close the connection to the ampqClient like you do here, or you can leverage the reference on the AMQPQueue interface:

```ts
function closeConnections(connections: DbConnections): Promise<void> {
  await connections.queue.channel.connection.close()
}
```

This way you can keep all resource cleanup in this single function and you can avoid leaking the abstraction provided by the queue interface.